### PR TITLE
LibWeb: Port {Custom,Mouse,UI,Wheel,}Event to new String

### PR DESCRIPTION
**Pull request by kennethmyhra**
All IDL implementations in the LibWeb event system is now ported to new String. There are still a few areas of the event system, not directly related to the IDL implementations, that is still using Deprecated{Fly}String.

### CODE REVIEW

File: Userland/Libraries/LibWeb/DOM/EventDispatcher.cpp [hunk 1]
```diff
@@ -181,13 +181,13 @@ void EventDispatcher::invoke(Event::PathEntry& struct_, Event& event, Event::Pha
...
+            event.set_type(HTML::EventNames::webkitanimationend);
         else if (event.type() == "animationiteration")
-            event.set_type("webkitAnimationIteration"sv);
+            event.set_type(HTML::EventNames::webkitanimationiteration);
         else if (event.type() == "animationstart")
-            event.set_type("webkitAnimationStart"sv);
+            event.set_type(HTML::EventNames::webkitanimationstart);
         else if (event.type() == "transitionend")
-            event.set_type("webkitTransitionEnd"sv);
+            event.set_type(HTML::EventNames::webkittransitionend);
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/DOM/EventDispatcher.cpp:190]: Hmm, don't we lose the weird casing here? i.e `webkitAnimationEnd` vs `webkitanimationend`.

Comment (kennethmyhra) [reply Userland/Libraries/LibWeb/DOM/EventDispatcher.cpp:190]: Oh yeah, I thought it didn't matter. But now I see they are legacy event types and that's probably why they have the weird casing.

Comment (kennethmyhra) [reply Userland/Libraries/LibWeb/DOM/EventDispatcher.cpp]: Fixed

**Comment by linusg** [comment]
LGTM but has conflicts

**Comment by kennethmyhra** [comment]
Conflicts fixed. I re-added the `pr-maintainer-approved-but-awaiting-ci` label :yakkie:

**Pull request merged**